sqlite.go

```diff
@@ -490,23 +490,22 @@ func (s *stmt) ExecContext(ctx context.Context, args []driver.NamedValue) (drive
 	if err := s.bindAll(args); err != nil {
 		return nil, s.reserr("Stmt.Exec(Bind)", err)
 	}
-	var done chan struct{}
+
+	// If non-nil, wait must be called prior to returning to ensure a
+	// cancellation context (if present) has completed, so that a cancellation
+	// cannot outlast this request and fire during a later execution.
+	var wait func()
 	if ctx.Value(queryCancelKey{}) != nil {
-		done = make(chan struct{})
+		done := make(chan struct{})
 		pctx, pcancel := context.WithCancel(ctx)
-		defer pcancel() // to make the AfterFunc fire and close(done)
-
 		db := s.stmt.DBHandle()
-		stop := context.AfterFunc(pctx, func() {
+		context.AfterFunc(pctx, func() {
 			defer close(done)
 			if ctx.Err() != nil {
```

Comment:
write a comment here why `ctx` and not `pctx`?

Reply:
Good idea, done (also in the other spot we do this).